zkevm-circuits/src/evm_circuit/op_execution/byte.rs

```diff
@@ -0,0 +1,292 @@
+                ExecutionStep {
+                    opcode: OpcodeId::BYTE,
+                    case: Case::Success,
+                    values: vec![index.clone(), value.clone(), result.clone()],
+                }
+            ],
+            vec![
+                Operation {
+                    gc: 1,
+                    target: Target::Stack,
+                    is_write: true,
+                    values: [
+                        Base::zero(),
+                        Base::from_u64(1023),
+                        compress(value.clone()),
+                        Base::zero(),
+                    ]
+                },
+                Operation {
+                    gc: 2,
+                    target: Target::Stack,
+                    is_write: true,
+                    values: [
+                        Base::zero(),
+                        Base::from_u64(1022),
+                        compress(index.clone()),
+                        Base::zero(),
+                    ]
+                },
+                Operation {
+                    gc: 3,
+                    target: Target::Stack,
+                    is_write: false,
+                    values: [
+                        Base::zero(),
+                        Base::from_u64(1022),
+                        compress(index),
+                        Base::zero(),
+                    ]
+                },
+                Operation {
+                    gc: 4,
+                    target: Target::Stack,
+                    is_write: false,
+                    values: [
+                        Base::zero(),
+                        Base::from_u64(1023),
+                        compress(value),
+                        Base::zero(),
+                    ]
+                },
+                Operation {
+                    gc: 5,
+                    target: Target::Stack,
+                    is_write: true,
+                    values: [
+                        Base::zero(),
+                        Base::from_u64(1023),
+                        compress(result),
+                        Base::zero(),
+                    ]
+                },
+            ],
+            Ok(())
+        );
+    }
+
+    #[test]
+    fn byte_gadget() {
+        // Select byte 29 (MSB is at 0)
+        check_byte_gadget(0x030201u64.into(), 29u64.into(), 0x03u64.into());
+        // Select byte 256
+        check_byte_gadget(0x030201u64.into(), 256u64.into(), 0u64.into());
+
+        // More extensive tests (which takes a long time to run, so disabled by default)
+        /*let test_value =
+            BigUint::from_bytes_be(&(1u8..33u8).collect::<Vec<u8>>()[..]);
+        for idx in 0u64..33 {
+            check_byte_gadget(
+                test_value.clone(),
+                idx.into(),
+                if idx < 32 {
+                    (idx + 1).into()
+                } else {
+                    0u64.into()
+                },
+            );
+        }*/
```

Comment:
Instead of leaving dead codes here, can we create an issue and put this code snippet there?

Reply:
Marked the exhaustive test as ignored so it doesn't run by default, but all ignored tests can be run when doing `cargo test -- --ignored`. Does that sound good to you? No more dead code and the test code is still nicely available next to the other code, and all future slow tests can do the same so there's just one command that can be run to all test these as well.